dissect/target/filesystem.py

```diff
@@ -58,26 +71,54 @@ def path(self, *args) -> fsutil.TargetPath:
         """Get a specific path from the filesystem."""
         return fsutil.TargetPath(self, *args)
 
+    @classmethod
+    def detect(cls, fh: BinaryIO) -> bool:
+        """Detect whether the ``fh`` file-handle is supported by this ``Filesystem`` implementation.
+
+        The position of ``fh`` will be restored before returning.
+
+        Args:
+            fh: A file-like object, usually a disk or partition.
+
+        Returns:
+            ``True`` if ``fh`` is supported, ``False`` otherwise.
+        """
+        offset = fh.tell()
+        try:
+            fh.seek(0)
```

Comment:
I noticed not all filesystem `detect()` implementations originally did this. Doing it anyway won't be a problem?

Reply:
Nope, it never really made sense that it didn't do it, because you could never "open" a filesystem from arbitrary locations anyway. The intended way to do that would be to use a `RelativeStream` or `RangeStream`.